src/OSPSuite.Core/Domain/ModelCoreSimulation.cs

```diff
@@ -27,13 +27,20 @@ public interface IModelCoreSimulation : IObjectBase, IWithCreationMetaData, IWit
       ///    on model creation for now. Adding <see cref="Reaction" /> to the building block will not change the model structure
       /// </summary>
       IReactionBuildingBlock Reactions { get; }
+
+      /// <summary>
+      ///    Name of all compounds used in the simulation
+      /// </summary>
+      IReadOnlyList<string> CompoundNames { get; }
+
+      IEnumerable<T> All<T>() where T : class, IEntity;
    }
 
    public class ModelCoreSimulation : ObjectBase, IModelCoreSimulation
    {
       public IModel Model { get; set; }
+
       public IBuildConfiguration BuildConfiguration { get; set; }
-      public virtual DataRepository Results { get; set; }
```

Comment:
Not sure why this is here of this is required. I think for MoBi but then I'll definie it in MoBi directly